Mathlib/Combinatorics/SimpleGraph/Girth.lean

```diff
@@ -9,46 +9,74 @@ import Mathlib.Data.ENat.Lattice
 -/
 
 namespace SimpleGraph
-variable {α : Type*} {G : SimpleGraph α} {n : ℕ∞}
+variable {α : Type*} {G : SimpleGraph α}
 
-/-- The girth of a simple graph is the length of its smallest cycle, or `∞` if the graph is acyclic.
+section egirth
+
+variable {n : ℕ∞}
+
+/--
+The extended girth of a simple graph is the length of its smallest cycle, or `∞` if the graph is
+acyclic.
 -/
-noncomputable def girth (G : SimpleGraph α) : ℕ∞ :=
-⨅ a, ⨅ w : G.Walk a a, ⨅ _ : w.IsCycle, w.length
+noncomputable def egirth (G : SimpleGraph α) : ℕ∞ :=
+  ⨅ a, ⨅ w : G.Walk a a, ⨅ _ : w.IsCycle, w.length
 
-@[simp] lemma le_girth : n ≤ G.girth ↔ ∀ a (w : G.Walk a a), w.IsCycle → n ≤ w.length := by
-  simp [girth]
+@[simp] lemma le_egirth : n ≤ G.egirth ↔ ∀ a (w : G.Walk a a), w.IsCycle → n ≤ w.length := by
+  rw [egirth, le_iInf_iff]
+  refine forall_congr' fun a ↦ le_iInf₂_iff
 
-@[simp] lemma girth_eq_top : G.girth = ⊤ ↔ G.IsAcyclic := by simp [girth, IsAcyclic]
+@[simp] lemma egirth_eq_top : G.egirth = ⊤ ↔ G.IsAcyclic := by simp [egirth, IsAcyclic]
 
-protected alias ⟨_, IsAcyclic.girth_eq_top⟩ := girth_eq_top
+protected alias ⟨_, IsAcyclic.egirth_eq_top⟩ := egirth_eq_top
 
-lemma girth_anti : Antitone (girth : SimpleGraph α → ℕ∞) :=
+lemma egirth_anti : Antitone (egirth : SimpleGraph α → ℕ∞) :=
   fun G H h ↦ iInf_mono fun a ↦ iInf₂_mono' fun w hw ↦ ⟨w.mapLe h, hw.mapLe _, by simp⟩
 
-lemma exists_girth_eq_length :
-    (∃ (a : α) (w : G.Walk a a), w.IsCycle ∧ G.girth = w.length) ↔ ¬ G.IsAcyclic := by
+lemma exists_egirth_eq_length :
+    (∃ (a : α) (w : G.Walk a a), w.IsCycle ∧ G.egirth = w.length) ↔ ¬ G.IsAcyclic := by
   refine ⟨?_, fun h ↦ ?_⟩
   · rintro ⟨a, w, hw, _⟩ hG
     exact hG _ hw
-  · simp_rw [← girth_eq_top, ← Ne.eq_def, girth, iInf_subtype', iInf_sigma', ENat.iInf_coe_ne_top,
+  · simp_rw [← egirth_eq_top, ← Ne.eq_def, egirth, iInf_subtype', iInf_sigma', ENat.iInf_coe_ne_top,
       ← exists_prop, Subtype.exists', Sigma.exists', eq_comm] at h ⊢
     exact ciInf_mem _
 
-lemma three_le_girth : 3 ≤ G.girth := by
+lemma three_le_egirth : 3 ≤ G.egirth := by
   by_cases h : G.IsAcyclic
-  · rw [← girth_eq_top] at h
+  · rw [← egirth_eq_top] at h
     rw [h]
     apply le_top
-  · rw [← exists_girth_eq_length] at h
+  · rw [← exists_egirth_eq_length] at h
     have ⟨_, _, _⟩ := h
     simp_all only [Nat.cast_inj, Nat.ofNat_le_cast, Walk.IsCycle.three_le_length]
 
-@[simp] lemma girth_bot : girth (⊥ : SimpleGraph α) = ⊤ := by simp
+@[simp] lemma egirth_bot : egirth (⊥ : SimpleGraph α) = ⊤ := by simp
+
+end egirth
+
+section girth
+
+variable {n : ℕ}
+
+/--
+The girth of a simple graph is the length of its smallest cycle, or junk value `0` if the graph is
+acyclic.
+-/
+noncomputable def girth (G : SimpleGraph α) : ℕ :=
+  G.egirth.toNat
+
+@[simp] lemma girth_bot : girth (⊥ : SimpleGraph α) = 0 := by
+  simp [girth]
+
+lemma three_le_girth (hG : ¬ G.IsAcyclic) : 3 ≤ G.girth :=
+  ENat.toNat_le_toNat three_le_egirth <| egirth_eq_top.not.mpr hG
```

Comment:
Please keep the two API in the same order
```suggestion
lemma three_le_girth (hG : ¬ G.IsAcyclic) : 3 ≤ G.girth :=
  ENat.toNat_le_toNat three_le_egirth <| egirth_eq_top.not.mpr hG

@[simp] lemma girth_bot : girth (⊥ : SimpleGraph α) = 0 := by
  simp [girth]
```

Reply:
I've had to change the order of some lemmas because of how they depend on each other. So for example, three_le_girth` ended up being before `girth_eq_zero`, although `three_le_egirth` is after `egirth_eq_top`. Is this okay?